src/arrays/compact.js

```diff
@@ -0,0 +1,5 @@
+function compact(array) {
+  return array.filter(Boolean);
+}
```

Comment:
Same prototype methd that have to be avoided:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter